Essentials/src/com/earth2me/essentials/services/ReserveEssentials.java

```diff
@@ -0,0 +1,1875 @@
+        this.ess = ess;
+    }
+
+    /**
+     * Register the provider if no other economy is available.
+     *
+     * @param ess The Essentials plugin instance
+     */
+    public static void register(Essentials ess) {
+        Reserve reserve = Reserve.instance();
+
+        // Check to see if there is an economy provider already registered.
+        // If so, the existing provider should take priority
+        if (!reserve.economyProvided()) {
+            Reserve.instance().registerProvider(new ReserveEssentials(ess));
+        }
+    }
+
+    /**
+     * @return The name of the Economy implementation.
+     */
+    @Override
+    public String name() {
+        return "Essentials Economy";
+    }
+
+    /**
+     * @return The version of Reserve the Economy implementation supports.
+     */
+    @Override
+    public String version() {
+        return "0.1.2.0";
+    }
+
+    //This is our method to convert UUID -> username for use with Essentials' create account methods.
+    private String getName(UUID identifier) {
+        final User user = ess.getUser(identifier);
+        return user == null ? identifier.toString() : user.getName();
+    }
+
+    /**
+     * @return Whether or not this implementation is enabled.
+     */
+    @Override
+    public boolean enabled() {
+        return true;
+    }
+
+    /**
+     * @return Whether or not this implementation should have a default Vault implementation.
+     */
+    @Override
+    public boolean vault() {
+        return false;
+    }
+
+    /**
+     * Used to get the plural name of the default currency.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultPlural() {
+        return "Dollars";
+    }
+
+    /**
+     * Used to get the singular name of the default currency.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultSingular() {
+        return "Dollar";
+    }
+
+    /**
+     * Used to get the plural name of the default currency for a world.
+     * @param world The world to be used in this check.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultPlural(String world) {
+        return "Dollars";
+    }
+
+    /**
+     * Used to get the singular name of the default currency for a world.
+     * @param world The world to be used in this check.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultSingular(String world) {
+        return "Dollar";
+    }
+
+    /**
+     * Checks to see if a {@link Currency} exists with this name.
+     * @param name The name of the {@link Currency} to search for.
+     * @return True if the currency exists, else false.
+     */
+    @Override
+    public boolean hasCurrency(String name) {
+        return true; //Always return true here as Essentials only supports one currency.
```

Comment:
Should this really be the case? This seems kind of sketchy - no matter what currency I try to use, Ess will have it but not really because all currencies are this one? Does Reserve have some additional UI on top of this which hides this detail for single currency economies?

Reply:
This goes back to the comment about Essentials not having a name for its currency, from what I've seen.

Reply:
I'm simply unfamiliar with how Reserve is used in this instance to know whether this is the correct behavior - should all currency be actually represented by the same bank even though they are different terms? Or should different terms error out because there is only one bank?

Reply:
I think I found a solution in the recent commits.